send-transaction-service/src/send_transaction_service.rs

```diff
@@ -8,25 +8,52 @@ use {
     solana_runtime::{bank::Bank, bank_forks::BankForks},
     solana_sdk::{hash::Hash, nonce_account, pubkey::Pubkey, signature::Signature},
     std::{
-        collections::hash_map::{Entry, HashMap},
+        collections::{
+            hash_map::{Entry, HashMap},
+            HashSet,
+        },
         net::SocketAddr,
-        sync::{Arc, RwLock},
-        thread::{self, Builder, JoinHandle},
+        sync::{
+            atomic::{AtomicBool, Ordering},
+            Arc, Mutex, RwLock,
+        },
+        thread::{self, sleep, Builder, JoinHandle},
         time::{Duration, Instant},
     },
 };
 
 /// Maximum size of the transaction queue
 const MAX_TRANSACTION_QUEUE_SIZE: usize = 10_000; // This seems like a lot but maybe it needs to be bigger one day
+
 /// Default retry interval
 const DEFAULT_RETRY_RATE_MS: u64 = 2_000;
+
 /// Default number of leaders to forward transactions to
 const DEFAULT_LEADER_FORWARD_COUNT: u64 = 2;
 /// Default max number of time the service will retry broadcast
 const DEFAULT_SERVICE_MAX_RETRIES: usize = usize::MAX;
 
+/// Default batch size for sending transaction in batch
+/// When this size is reached, send out the transactions.
+const DEFAULT_TRANSACTION_BATCH_SIZE: usize = 1;
+
+// The maximum transaction batch size
+pub const MAX_TRANSACTION_BATCH_SIZE: usize = 10_000;
+
+/// Maximum transactions per second
+pub const MAX_TPS: u64 = 1_000;
```

Comment:
Is the intent to limit how many tx per sec can be sent via an RPC server? Maybe we can name this constant better, as overall MAX TPS should be higher than 1K.

Reply:
We are not doing broadcasting here -- broadcast implies sending to multiple destinations simultaneously. I will change the name to MAX_TRANSACTION_SEND_PER_SECOND.